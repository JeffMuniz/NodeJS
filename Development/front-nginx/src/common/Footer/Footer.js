import React from "react";
import { Row, Col } from "react-styled-flexboxgrid";
import {
  FooterCol,
  FooterRow,
  Text,
  TextHeader,
  Attendance,
  Sac,
  Ombudsman,
  AttendanceText,
  AttendanceCol,
  Line,
  Copyright,
  SantanderDiv,
  Icon,
} from "./Footer.styles";

const Footer = () => (
  <FooterRow>
    <FooterCol xsOffset={2} xs={10}>
      <Row>
        <Col xs={12}>
          <AttendanceText>Atendimento</AttendanceText>
        </Col>
      </Row>
      <Row>
        <AttendanceCol xs={12} md={6}>
          <Icon name="attendance" /> Atendimento on-line
        </AttendanceCol>
        <SantanderDiv xs={12} md={6}>
          Uma empresa <Icon name="macnabank" />
        </SantanderDiv>
      </Row>
      <Row>
        <Col xs={12}>
          <Attendance>
            <Icon name="telphone" />
            <TextHeader>Atendimento: (11) 3456 0909</TextHeader>
            <Text>
              Disponível de segunda a sexta-feira das 8:00 as 20:00 horas,
              exceto em feriados nacionais.
            </Text>
          </Attendance>
          <Sac>
            <TextHeader>SAC: 0800 762 7777</TextHeader>
            <Text>Disponível 24 horas por dia, 7 dias por semana.</Text>
          </Sac>
          <Ombudsman>
            <TextHeader>OUVIDORIA: 0800 726 0322</TextHeader>
            <Text>
              Disponível de segunda a sexta-feira, das 9h às 18h, exceto
              feriados.
            </Text>
          </Ombudsman>
        </Col>
      </Row>
    </FooterCol>
    <Col xs={12}>
      <Line />
      <Copyright>
        <Text>Copyright Grupo macnabank© 2018</Text>
        <Text>
          For permission to reprint articles from this site, please contact:
          mac@comvc.com.br
        </Text>
      </Copyright>
    </Col>
  </FooterRow>
);

export default Footer;
