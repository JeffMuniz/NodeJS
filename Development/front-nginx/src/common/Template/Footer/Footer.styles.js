import styled from "styled-components";
import { Col, Row } from "react-styled-flexboxgrid";
import BaseIcon from "src/common/Icon/Icon";
import {
  grey,
  lighterBlack,
  rgbaLightGrey,
  rgbaShadowGrey,
  dustyGray,
} from "src/styles/colors";

export const Icon = styled(BaseIcon)`
  & > path {
    fill: ${dustyGray};
  }
`;

const ColBase = styled(Col)`
  color: ${grey};
`;

const PBase = styled.p`
  font-style: normal;
  letter-spacing: 0.1px;
  font-weight: normal;
`;

export const AttendanceCol = styled(ColBase)`
  margin-bottom: 15px;
`;

export const IconCol = styled(ColBase)`
  margin-bottom: 15px;
  text-align: center;
`;

export const FooterCol = styled(ColBase)`
  margin-top: 152px;
`;

export const FooterRow = styled(Row)`
  height: 815px;
`;

export const AttendanceText = styled.span`
  font-weight: bold;
  line-height: 26px;
  font-size: 16px;
  color: ${lighterBlack};
`;

export const Text = styled(PBase)`
  margin: 0;
  line-height: 16px;
  font-size: 12px;
`;

export const PhoneText = styled(PBase)`
  margin: 7px 0 0 0;
  line-height: 18px;
  font-size: 14px;

  small {
    font-size: 12px;
    letter-spacing: 0.1px;
  }
`;

export const TextItalic = styled(PBase)`
  margin: 7px 0 20px 0;
  letter-spacing: 0.1px;
  font-size: 13px;
  font-style: italic;
  color: ${dustyGray};
`;

export const SmallGrayText = styled.small`
  margin: 7px 0 20px 0;
  letter-spacing: 0.1px;
  font-size: 14px;
  color: ${dustyGray};
`;

export const Line = styled.div`
  background: ${rgbaShadowGrey};
  height: 3px;
  box-shadow: inset 0px 1px 0px ${rgbaLightGrey};
`;

export const Copyright = styled.div`
  margin-top: 30px;
  text-align: center;
  color: ${grey};
  ${Text} {
    line-height: 22px;
  }
`;

export const SantanderDiv = styled(ColBase)`
  align-items: flex-end;
  display: flex;
  font-size: 15px;
  margin-top: 26px;
  & > img {
    margin-left: 5px;
  }
`;
