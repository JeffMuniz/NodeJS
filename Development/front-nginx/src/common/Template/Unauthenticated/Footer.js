import React, { Fragment } from "react";
import {
  FooterContent,
  FooterH1,
  FooterImage,
  FooterInfo,
  FooterSocialMedia,
  FooterText,
} from "./Footer.styles";

const Footer = () => (
  <Fragment>
    <FooterContent>
      <FooterInfo>
        <FooterH1>mac macefícios e Serviços S.A.</FooterH1>
        <FooterText>
          Rua Amador Bueno, 474, Bloco I Andar 2 CEP 04752-901, São Paulo/SP •
          CNPJ/MF nº 30.798.783/0001-61
        </FooterText>
      </FooterInfo>
      <FooterSocialMedia />
    </FooterContent>
    <FooterImage />
  </Fragment>
);

export default Footer;
