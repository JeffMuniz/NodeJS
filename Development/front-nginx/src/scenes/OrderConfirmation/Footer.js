import React from "react";
import PropTypes from "prop-types";
import { Footer, Button } from "./styles";

const FooterWrapper = ({ handleCancel, handleConfirm }) => (
  <Footer>
    <Button
      id="button_cancel_order"
      onPress={handleCancel}
      value="cancelar pedido"
      buttonType="light"
    />
    <Button
      id="button_confirmation_order"
      onPress={handleConfirm}
      value="confirmar pedido"
      buttonType="warning"
    />
  </Footer>
);

FooterWrapper.propTypes = {
  handleCancel: PropTypes.func.isRequired,
  handleConfirm: PropTypes.func.isRequired,
};

export default FooterWrapper;
