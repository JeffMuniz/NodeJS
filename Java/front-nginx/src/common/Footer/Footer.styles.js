import styled from "styled-components";
import { Col, Row } from "react-styled-flexboxgrid";
import BaseIcon from "../Icon/Icon";
import {
  grey,
  lighterBlack,
  rgbaLightGrey,
  rgbaShadowGrey,
} from "../../styles/colors";

export const Icon = styled(BaseIcon)``;

const ColBase = styled(Col)`
  color: ${grey};
`;

const PBase = styled.p`
  font-style: normal;
  letter-spacing: 0.1px;
  font-weight: normal;
`;

const DivBase = styled.div`
  margin: 10px 0 10px 35px;
`;

export const FooterCol = styled(ColBase)`
  margin-top: 152px;
`;

export const FooterRow = styled(Row)`
  height: 815px;
`;

export const AttendanceCol = styled(ColBase)`
  display: flex;
  align-items: flex-end;
  ${Icon} {
    margin-right 10px;
  }
`;

export const AttendanceText = styled(PBase)`
  font-weight: bold;
  line-height: 26px;
  font-size: 16px;
  color: ${lighterBlack};
`;

export const Text = styled(PBase)`
  margin: 0;
  line-height: 16px;
  font-size: 14px;
`;

export const TextHeader = styled(PBase)`
  margin: 0;
  line-height: 16px;
  font-size: 15px;
`;

export const Sac = styled(DivBase)`
  width: 230px;
`;

export const Ombudsman = styled(DivBase)`
  width: 290px;
`;

export const Attendance = styled(DivBase)`
  width: 285px;
  position: relative;
  ${Icon} {
    position: absolute;
    right: 292px;
    top: 0;
  }
`;

export const Line = styled.div`
  background: ${rgbaShadowGrey};
  height: 3px;
  box-shadow: inset 0px 1px 0px ${rgbaLightGrey};
`;

export const Copyright = styled.div`
  margin-top: 30px;
  text-align: center;
  color: ${grey};
  ${Text} {
    line-height: 22px;
  }
`;

export const SantanderDiv = styled(ColBase)`
  display: flex;
  align-items: flex-end;
  ${Icon} {
    margin-left: 5px;
  }
`;
