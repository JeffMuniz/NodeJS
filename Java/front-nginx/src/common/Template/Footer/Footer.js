import React from "react";
import { Row, Col } from "react-styled-flexboxgrid";

import { logoBanco } from "@assets";
import {
  FooterCol,
  FooterRow,
  Text,
  PhoneText,
  TextItalic,
  AttendanceText,
  AttendanceCol,
  Line,
  Copyright,
  SantanderDiv,
  IconCol,
  Icon,
  SmallGrayText,
} from "./Footer.styles";

const Footer = () => (
  <FooterRow>
    <FooterCol xs={12}>
      <Row>
        <Col xs={12} mdOffset={2} md={4}>
          <Row>
            <AttendanceCol xs={12}>
              <AttendanceText>Atendimento</AttendanceText>
            </AttendanceCol>
            <IconCol xs={2}>
              <Icon name="telephone" />
            </IconCol>
            <AttendanceCol xs={10}>
              <PhoneText>
                4004 4474 <small>(Capitais e Regiões Metropolitanas)</small>
              </PhoneText>
              <PhoneText>
                0800 723 4474 <small>(Demais localidades)</small>
              </PhoneText>
              <TextItalic>
                Disponível de segunda à sexta, das 8h às 20h, exceto em feriados
                nacionais.
              </TextItalic>
            </AttendanceCol>
            <AttendanceCol xs={10} xsOffset={2}>
              <div>
                <PhoneText>
                  <strong>SAC</strong>
                </PhoneText>
                <PhoneText>
                  0800 723 5013 <SmallGrayText>para todo brasil</SmallGrayText>
                </PhoneText>
                <PhoneText>
                  0800 723 5014{" "}
                  <SmallGrayText>
                    para deficientes auditivos e pessoas com disturbio de fala
                  </SmallGrayText>
                </PhoneText>
              </div>
            </AttendanceCol>
          </Row>
        </Col>
        <Col xs={12} md={3} mdOffset={1}>
          <Row>
            <SantanderDiv xs={12}>
              Uma empresa <img src={logoBanco} alt="macnabank" />
            </SantanderDiv>
          </Row>
        </Col>
      </Row>
    </FooterCol>
    <Col xs={12}>
      <Line />
      <Copyright>
        <Text>Copyright Grupo macnabank© 2018</Text>
      </Copyright>
    </Col>
  </FooterRow>
);

export default Footer;
