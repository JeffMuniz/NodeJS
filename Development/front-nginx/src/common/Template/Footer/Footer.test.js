import React from "react";
import ShallowRenderer from "react-test-renderer/shallow";

import Footer from "./Footer.js";

describe("Footer", () => {
  it("should render it when not logged", () => {
    // given
    const renderer = new ShallowRenderer();

    // when
    renderer.render(
      <Footer>
        <div id="test">Test</div>
      </Footer>,
    );
    const result = renderer.getRenderOutput();

    // then
    expect(result).toMatchSnapshot();
  });
});
