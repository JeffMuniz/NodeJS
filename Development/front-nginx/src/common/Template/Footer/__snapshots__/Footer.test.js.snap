// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer should render it when not logged 1`] = `
<Styled(Row)>
  <Styled(Styled(Col))
    xs={12}
  >
    <Row>
      <Col
        md={4}
        mdOffset={2}
        xs={12}
      >
        <Row>
          <Styled(Styled(Col))
            xs={12}
          >
            <styled.span>
              Atendimento
            </styled.span>
          </Styled(Styled(Col))>
          <Styled(Styled(Col))
            xs={2}
          >
            <Styled(Icon)
              name="telephone"
            />
          </Styled(Styled(Col))>
          <Styled(Styled(Col))
            xs={10}
          >
            <Styled(styled.p)>
              4004 4474 
              <small>
                (Capitais e Regiões Metropolitanas)
              </small>
            </Styled(styled.p)>
            <Styled(styled.p)>
              0800 723 4474 
              <small>
                (Demais localidades)
              </small>
            </Styled(styled.p)>
            <Styled(styled.p)>
              Disponível de segunda à sexta, das 8h às 20h, exceto em feriados nacionais.
            </Styled(styled.p)>
          </Styled(Styled(Col))>
          <Styled(Styled(Col))
            xs={10}
            xsOffset={2}
          >
            <div>
              <Styled(styled.p)>
                <strong>
                  SAC
                </strong>
              </Styled(styled.p)>
              <Styled(styled.p)>
                0800 723 5013 
                <styled.small>
                  para todo brasil
                </styled.small>
              </Styled(styled.p)>
              <Styled(styled.p)>
                0800 723 5014
                 
                <styled.small>
                  para deficientes auditivos e pessoas com disturbio de fala
                </styled.small>
              </Styled(styled.p)>
            </div>
          </Styled(Styled(Col))>
        </Row>
      </Col>
      <Col
        md={3}
        mdOffset={1}
        xs={12}
      >
        <Row>
          <Styled(Styled(Col))
            xs={12}
          >
            Uma empresa 
            <img
              alt="macnabank"
              src={
                Object {
                  "testUri": "../../../src/assets/pngs/logo-banco.png",
                }
              }
            />
          </Styled(Styled(Col))>
        </Row>
      </Col>
    </Row>
  </Styled(Styled(Col))>
  <Col
    xs={12}
  >
    <styled.div />
    <styled.div>
      <Styled(styled.p)>
        Copyright Grupo macnabank© 2018
      </Styled(styled.p)>
    </styled.div>
  </Col>
</Styled(Row)>
`;
